tests/integration/test_scale.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests."""
+
+import asyncio
+import json
+import logging
+import time
+
+import ops
+import pytest
+from pytest_operator.plugin import Model, OpsTest
+
+import models
+import tests.integration.helpers
+
+logger = logging.getLogger(__name__)
+
+
+async def deploy_any_charm(  # pylint: disable=too-many-arguments
+    app: ops.model.Application,
+    ops_test: OpsTest,
+    model: Model,
+    any_app_number: int,
+    machines: list[str] | None,
+    entries: list[models.DnsEntry],
+):
+    """Deploy any charm and relate it to the bind-operator.
+
+    Args:
+        app: Deployed bind-operator app
+        ops_test: The ops test framework instance
+        model: The ops model instance
+        any_app_number: Number of the to be deployed any-charm
+        machines: The machines to deploy the any-charm onto
+        entries: List of DNS entries for any-charm
+    """
+    anyapp_name = f"anyapp-t{any_app_number}"
+    if machines is not None:
+        machine = machines[any_app_number % len(machines)]
+    else:
+        machine = None
+    logger.info("Deploying %s on %s", anyapp_name, machine)
+    if anyapp_name in model.applications:
+        return
+    await tests.integration.helpers.generate_anycharm_relation(
+        app,
+        ops_test,
+        anyapp_name,
+        entries,
+        machine=machine,
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.skip(reason="Scaling test")
```

Comment:
Maybe running it in the pipeline (even if it is a "lighter" version)  so the test is maintained?

Reply:
I don't see the point of a lighter version. It's not testing functionality, it is only used for deploying the stuff and then we can decide if we like the result

Reply:
I was just thinking that code that does not run, will stop working... but sure, close the comment